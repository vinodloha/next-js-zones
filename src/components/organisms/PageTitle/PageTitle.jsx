import React from 'react'
import PropTypes from 'prop-types'
import Head from 'next/head'

const PageTitle = ({ title }) => {
  return (
    <Head>
      <title key="title">{title}</title>
    </Head>
  )
}

PageTitle.propTypes = {
  title: PropTypes.string.isRequired
}

export default PageTitle
