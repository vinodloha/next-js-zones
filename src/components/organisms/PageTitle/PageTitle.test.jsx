import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import PageTitle from '.'

describe('PageTitle', () => {
  // Test uses ShallowRenderer.render() instead of TestRenderer.create()
  // because <PageTitle> encapsulates NextJS <Head> which in turn uses
  // React <Helmet> and Helmet doesn't show up in a normal component render.
  it('creates a <head> element suitable for global usage across all pages', () => {
    const renderer = new ShallowRenderer()
    renderer.render(<PageTitle title="mock title for test" />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
  it('populates the specified page title', () => {
    const title = 'mockTitle'
    const renderer = new ShallowRenderer()
    renderer.render(<PageTitle title={title} />)
    const result = renderer.getRenderOutput()
    // Expect a React module with a populated <title>
    expect(result.props.children).toEqual(
      expect.objectContaining({
        type: 'title',
        props: {
          children: title
        }
      })
    )
  })
})
