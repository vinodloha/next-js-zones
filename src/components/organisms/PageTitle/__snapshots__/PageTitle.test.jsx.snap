// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle creates a <head> element suitable for global usage across all pages 1`] = `
<Head>
  <title>
    mock title for test
  </title>
</Head>
`;
